### PR TITLE
feat(calendar): calendar starts on correct day of week

### PR DESCRIPTION
**Pull request by triedal**
[Card](https://rentpath.atlassian.net/browse/AG-2676)

### CODE REVIEW

File: packages/react-ui-core/src/DatePicker/Calendar.js [hunk 1]
```diff
@@ -67,9 +67,9 @@ export default class Calendar extends PureComponent {
     // NOTE: consitently create 5 rows so the calendar pads shorter months
     for (let i = 0; i <= 5; i += 1) {
       days.push(
-        <div key={`date-row-${i}`}>
+        <Fragment key={`date-row-${i}`}>
```
Comment (tadjohnston) [inline packages/react-ui-core/src/DatePicker/Calendar.js]: Why did you change this to a fragment?

Comment (triedal) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: How does this affect styling? It has no classname

Comment (tadjohnston) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: There's no dom element it's attached to. Did you check the output. It probably looks like this based on your change:
```
<div class="Calendar_DaysOfWeek">SuMoTuWeThFriSa</div>
```

Comment (tadjohnston) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: Sorry, I was thinking the wrong thing. But by removing this, it would cause there to be no rows which I assume you could do something with flex to address it, but it might be more difficult than just leaving the div.

File: packages/react-ui-core/src/DatePicker/Calendar.js [hunk 2]
```diff
@@ -43,6 +48,34 @@ export default class Calendar extends PureComponent {
...
+    return daysOfWeek
+  }
+
+  get daysOfMonth() {
+    const days = []
+
+    // NOTE: consitently create 5 rows so the calendar pads shorter months
+    for (let i = 0; i <= 5; i += 1) {
+      days.push(
+        <div key={`date-row-${i}`}>
```
Comment (barelyfitz) [inline packages/react-ui-core/src/DatePicker/Calendar.js]: I don't think there is any need to make the key unique, why not just `key={i}`

Comment (barelyfitz) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: Oh, maybe this array gets concatenated with this other, so you might need that afterall. If so please ignore.

Comment (barelyfitz) [inline packages/react-ui-core/src/DatePicker/Calendar.js]: Since the normal dates of the month are all `button` elements (an inline element) this should probably also be an inline element (`span` instead of a `div`). For example, look at the HTML without styling, you don't want the days to appear one per line, you want them to appear all on the same line like the other dates.

**Comment by barelyfitz** [comment]
Storybook needs to be updated to add styles:
Storybook

File: packages/react-ui-core/src/DatePicker/Calendar.js (hunk 2 above)
Comment (triedal) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: I believe this should be a block level element as it contains an entire week, not just a day.

Comment (tadjohnston) [reply packages/react-ui-core/src/DatePicker/Calendar.js]: I think you're right on this one @triedal. I would change the days of the week though. I'll try to name some of this stuff better in the ui-ux version and it's somewhat confusing lol.
